prboom2/src/st_stuff.c

```diff
@@ -791,15 +791,26 @@ static void ST_doPaletteStuff(void)
 
   if (cnt)
     {
-      palette = (cnt+7)>>3;
-      if (palette >= NUMREDPALS)
-        palette = NUMREDPALS-1;
+      // In Chex Quest, the player never sees red.  Instead, the
+      // radiation suit palette is used to tint the screen green,
+      // as though the player is being covered in goo by an
+      // attacking flemoid.
+      if (gamemission == chex)
```

Comment:
We should still have a check if the pain palette is enabled right?

Reply:
If it wasn't, we wouldn't even enter the `if (cnt)` block.

Reply:
Ah right